akita/src/internal/crud.ts

```diff
@@ -69,16 +69,28 @@ export class CRUD {
     };
   }
 
-  _update<T extends EntityState>(state: T, ids: ID[], newStateOrFn: object | ((e: Readonly<any>) => object)): T {
+  _update<T extends EntityState>(state: T, ids: ID[], newStateOrFn: object | ((e: Readonly<any>) => object), idKey: string): T {
     const updatedEntities = {};
 
+    let isUpdatingIdKey = false;
+    let idToUpdate: ID;
+
     for (let i = 0; i < ids.length; i++) {
       const id = ids[i];
+      idToUpdate = id;
       assertEntityExists(id, state.entities);
 
       const oldEntity = state.entities[id];
       const newState = isFunction(newStateOrFn) ? newStateOrFn(oldEntity) : newStateOrFn;
 
+      if (newState.hasOwnProperty(idKey)) {
```

Comment:
You can't check against `newState`. Your update method could look like this:
```ts
 this.store.update(1, entity => {
   return {
      ...entity,
      id: 5,
      completed: !entity.completed
   }
})
```

Reply:
that is a valid point.
Just curious though, The following would achieve the same result, right?
```ts
this.store.update(1, entity => {
   return {
      completed: !entity.completed
   }
})
```

Reply:
So I guess I also need to check if the value of id field has changed. Something like:
```ts
if (newState.hasOwnProperty(idKey) && newState[idKey] !== oldEntity[idKey]) {
```

Reply:
Yes but you cant force it.

Reply:
Yes. This seems right.